p/db/index.js

```diff
@@ -23,12 +27,120 @@ const transformReqBody = (ctx, body) => {
 	body.ext = 'DB.R19.04.a'
-	body.ver = '1.16'
+	body.ver = '1.15'
 	body.auth = {type: 'AID', aid: 'n91dB8Z77MLdoR0K'}
 
 	return body
 }
 
+const slices = (n, arr) => {
+	const initialState = {slices: [], count: Infinity}
+	return arr.reduce(({slices, count}, item) => {
+		if (count >= n) {
+			slices.push([item])
+			count = 1
+		} else {
+			slices[slices.length - 1].push(item)
+			count++
+		}
+		return {slices, count}
+	}, initialState).slices
+}
+
+const parseGrid = (g) => {
+	// todo: g.type, e.g. `S`
+	return {
+		title: g.title,
+		rows: slices(g.nCols, g.itemL.map(item => item.msgL[0]))
+	}
+}
+
+// e.g. `#NULL:0`, `#VGN:3020010`
+const parseLocalId = (str) => {
+	if (!str || str.slice(0, 6) === '#NULL:') return null
+	const [network, id] = str.split(':')
+	return [network.slice(1), id]
+}
+
+const ausstattungKeys = Object.assign(Object.create(null), {
+	'3-s-zentrale': '3SZentrale',
+	'parkplatze': 'parkingLots',
+	'fahrrad-stellplatze': 'bicycleParkingRacks',
+	'opnv-anbindung': 'localPublicTransport',
+	'wc': 'toilets',
+	'schliessfacher': 'lockers',
+	'reisebedarf': 'travelShop',
+	'stufenfreier-zugang': 'stepFreeAccess',
+	'ein-umsteigehilfe': 'boardingAid',
+	'taxi-am-bahnhof': 'taxis'
+})
+const parseAusstattungVal = (val) => {
+	val = val.toLowerCase()
+	return val === 'ja' ? true : (val === 'nein' ? false : val)
+}
+
+const parseAusstattungGrid = (g) => {
+	// filter duplicate hint rows
+	const rows = uniqBy(g.rows, ([key, val]) => key + ':' + val)
+
+	const res = {raw: rows}
+	for (let [key, val] of rows) {
+		key = ausstattungKeys[slugg(key)]
+		if (key) res[key] = parseAusstattungVal(val)
+	}
+	return res
+}
+
+const parseReisezentrumÖffnungszeiten = (g) => {
+	const res = {}
+	for (const [dayOfWeek, val] of g.rows) res[dayOfWeek] = val
+	res.raw = g.rows
+	return res
+}
+
+const parseLocWithDetails = ({parsed, common}, l) => {
+	if (!parsed) return parsed
+	if (parsed.type !== 'stop' && parsed.type !== 'station') return parsed
+
+	if (Array.isArray(l.gridL)) {
+		const resolveCell = cell => 'hint' in cell ? cell.hint.text : cell
+		const resolveCells = grid => ({
+			...grid,
+			rows: grid.rows.map(row => row.map(resolveCell))
+		})
+
+		let grids = l.gridL
+		.map(grid => parseGrid(grid, common))
+		.map(resolveCells)
+
+		const ausstattung = grids.find(g => slugg(g.title) === 'ausstattung')
+		if (ausstattung) {
+			parsed.facilities = parseAusstattungGrid(ausstattung)
+		}
+		const öffnungszeiten = grids.find(g => slugg(g.title) === 'offnungszeiten-reisezentrum')
+		if (öffnungszeiten) {
+			parsed.reisezentrumOpeningHours = parseReisezentrumÖffnungszeiten(öffnungszeiten)
+		}
+
+		grids = without(grids, ausstattung, öffnungszeiten)
+		if (grids.length > 0) parsed.grids = grids
+	}
+
+	const hints = (l.msgL || [])
+	.map(ref => ref.hint)
+	.filter(hint => !!hint)
+
+	// todo: is this the DHID?
+	const dhid = hints.find(h => h.code === 'IF')
+	if (dhid) parsed.dhid = dhid.text
```

Comment:
On `master`, this is being parsed by `parseLocation` already, as `stop.ids.dhid`. I should remove it here.